libstuff/SData.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include "libstuff.h"
+
+// --------------------------------------------------------------------------
+// A very simple HTTP-like structure consisting of a method line, a table,
+// and a content body.
+// --------------------------------------------------------------------------
+struct SData {
+    // Public attributes
+    string methodLine;
+    STable nameValueMap;
+    string content;
+
+    // Constructors
+    SData();
+
+    // Initializes a new SData from a string.  If the string provided is not
+    // an entire HTTPs like message, the string is used as the methodLine.
+    SData(const string& fromString);
+
+    // Allow forwarding emplacements directly so SData can act like `std::map`.
+    template <typename... Ts>
+    pair<decltype(nameValueMap)::iterator, bool> emplace(Ts&&... args) {
+        return nameValueMap.emplace(forward<Ts>(args)...);
+    }
+
+
+    // Operators
+    // This version creates an entry, if necessary, and returns a reference
+    string& operator[](const string& name);
+
+    // This version takes care not to create an entry if none is present
+    const string& operator[](const string& name) const;
+
+    // Two templated versions of `set` are provided. One for arithmetic types, and one for other types (which must be
+    // convertible to 'string'). These allow you to do the following:
+    // SData.set("count", 7);
+    // SData.set("name", "Tyler");
+    // for all string and integer types.
+    template <typename T>
+    typename enable_if<is_arithmetic<T>::value, void>::type set(const string& key, const T item)
+    {
+        nameValueMap[key] = to_string(item);
+    }
+    template <typename T>
+    typename enable_if<!is_arithmetic<T>::value, void>::type set(const string& key, const T item)
+    {
+        nameValueMap[key] = item;
+    }
+
+    // Mutators
+    // Erase everything
+    void clear();
+
+    // Erase one value
+    void erase(const string& name);
+
+    // Combine the name/value pairs from two datas
+    void merge(const STable& table);
+
+    // Combine two SData into one
+    void merge(const SData& rhs);
+
+    // Accessors
+    // Returns whether or not this data is empty
+    bool empty() const;
+
+    // Returns whether or not a particular value has been set
+    bool isSet(const string& name) const;
+
+    // Return as a 32-bit value.
```

Comment:
This value is probably 32-bit on most machines.

Reply:
made this explicitly 32bits, compiled everything, didn't seen to break anywhere.